fairscale/optim/adam.py

```diff
@@ -62,20 +67,24 @@ def __init__(
             weight_decay: Optional[float] = 0.0,
             max_grad_norm: Optional[float] = 0.0,
             amsgrad: Optional[bool] = False,
-            mixed_precision: Optional[bool] = False,
-            optim_fp16: Optional[bool] = False,
+            precision: Optional[Union[None, Precision]] = None,
```

Comment:
Isn't `Optionsal[Precision]` enough? Why do we need `Union[None,Precision]`?

Reply:
Yes, you're totally right! I thought that I needed to allow None since the default is None, but I was mistaken.